File: bubbleSort.py
"""Bubble sort implementation."""

def bubble_sort(data: list, reverse: bool = False) -> list:
    """Bubble sorting function.
    Args:
        data (list): list to be ordered.
        reverse (bool, optional): ascending(False) or descending(True) order. Defaults to False.
    Returns:
        list: ordered list.
    """
    n = len(data)
    # 1st loop goes through the list as many times as there are elements
    for i in range(n):
        for j in range(n - 1):
            if data[j] > data[j + 1]:
                data[j], data[j + 1] = data[ j + 1], data[j]
    if reverse:
        data = data[::-1]
    return data

File: bubbleSort.ts
/* Bubble sort implementation. */

function bubble_sort(data: number[], reverse: boolean=false): number[] {
  /**
   * Bubble sorting function.
   * Args:
      data number[]: list to be ordered.
        reverse (bool, optional): ascending(False) or descending(True) order. Defaults to False.
    Returns:
        number[]: ordered list.
   */
  const n: number = data.length;
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n - 1; j++) {
      if (data[j] > data[j + 1]) {
        let temp = data[j];
        data[j] = data[j + 1]
        data[j + 1] = temp
      }
    }
  }
  if (reverse) {
    let newArray: number[] = [];
    // this "for" is used to avoid the use of the array.reverse() built in function
    for (let i = 0; i < n; i++) {
      newArray[i] = data[n - i];
    }
    return newArray;
  }
  return data
}

